error: unknown type name 'Array_fixed_byte_512'

V full version: V 0.2.2 c1d4074
OS: linux, "Void Linux"
Processor: 32 cpus, 64bit, little endian, ppc64le
CC version: cc (GCC) 10.2.1 20201203

`array_fixed_byte.v` is:
```v
type Block = [512]byte
```

Result is:
```
$ v -cg array_fixed_byte.v 
/tmp/v/array_fixed_byte.17656917884316645732.tmp.c:721:9: error: unknown type name 'Array_fixed_byte_512'
  721 | typedef Array_fixed_byte_512 main__Block;
      |         ^~~~~~~~~~~~~~~~~~~~
builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please raise an issue on GitHub:
...
```

Expected result is that `Block` is a new type name for a block of 512 bytes.
If there is something I can do to help solve this please let me know. I’m in Discord too.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -760,6 +760,32 @@ pub fn (mut g Gen) write_typedef_types() {
 			.array {
 				g.type_definitions.writeln('typedef array $typ.cname;')
 			}
+			.array_fixed {
+				info := typ.info as table.ArrayFixed
+				elem_sym := g.table.get_type_symbol(info.elem_type)
+				if elem_sym.is_builtin() {
+					// .array_fixed {
+					styp := typ.cname
+					// array_fixed_char_300 => char x[300]
+					mut fixed := styp[12..]
+					len := styp.after('_')
+					fixed = fixed[..fixed.len - len.len - 1]
+					if fixed.starts_with('C__') {
+						fixed = fixed[3..]
+					}
+					if elem_sym.info is table.FnType {
+						pos := g.out.len
+						g.write_fn_ptr_decl(&elem_sym.info, '')
+						fixed = g.out.after(pos)
+						g.out.go_back(fixed.len)
+						mut def_str := 'typedef $fixed;'
+						def_str = def_str.replace_once('(*)', '(*$styp[$len])')
+						g.type_definitions.writeln(def_str)
+					} else {
+						g.type_definitions.writeln('typedef $fixed $styp [$len];')
+					}
+				}
+			}
 			.interface_ {
 				g.write_interface_typesymbol_declaration(typ)
 			}
@@ -5430,16 +5456,18 @@ fn (mut g Gen) write_types(types []table.TypeSymbol) {
 				}
 				elem_type := typ.info.elem_type
 				elem_sym := g.table.get_type_symbol(elem_type)
-				if elem_sym.info is table.FnType {
-					pos := g.out.len
-					g.write_fn_ptr_decl(&elem_sym.info, '')
-					fixed = g.out.after(pos)
-					g.out.go_back(fixed.len)
-					mut def_str := 'typedef $fixed;'
-					def_str = def_str.replace_once('(*)', '(*$styp[$len])')
-					g.type_definitions.writeln(def_str)
-				} else {
-					g.type_definitions.writeln('typedef $fixed $styp [$len];')
+				if !elem_sym.is_builtin() {
+					if elem_sym.info is table.FnType {
+						pos := g.out.len
+						g.write_fn_ptr_decl(&elem_sym.info, '')
+						fixed = g.out.after(pos)
+						g.out.go_back(fixed.len)
+						mut def_str := 'typedef $fixed;'
+						def_str = def_str.replace_once('(*)', '(*$styp[$len])')
+						g.type_definitions.writeln(def_str)
+					} else {
+						g.type_definitions.writeln('typedef $fixed $styp [$len];')
+					}
 				}
 			}
 			else {}

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5445,18 +5445,17 @@ fn (mut g Gen) write_types(types []table.TypeSymbol) {
 				g.type_definitions.writeln('')
 			}
 			table.ArrayFixed {
-				// .array_fixed {
-				styp := typ.cname
-				// array_fixed_char_300 => char x[300]
-				mut fixed := styp[12..]
-				len := styp.after('_')
-				fixed = fixed[..fixed.len - len.len - 1]
-				if fixed.starts_with('C__') {
-					fixed = fixed[3..]
-				}
-				elem_type := typ.info.elem_type
-				elem_sym := g.table.get_type_symbol(elem_type)
+				elem_sym := g.table.get_type_symbol(typ.info.elem_type)
 				if !elem_sym.is_builtin() {
+					// .array_fixed {
+					styp := typ.cname
+					// array_fixed_char_300 => char x[300]
+					mut fixed := styp[12..]
+					len := styp.after('_')
+					fixed = fixed[..fixed.len - len.len - 1]
+					if fixed.starts_with('C__') {
+						fixed = fixed[3..]
+					}
 					if elem_sym.info is table.FnType {
 						pos := g.out.len
 						g.write_fn_ptr_decl(&elem_sym.info, '')

--- a/vlib/v/tests/alias_fixed_array_test.v
+++ b/vlib/v/tests/alias_fixed_array_test.v
@@ -1,6 +1,6 @@
 type Block = [8]byte
 
-fn test_alias_fixed_array(){
+fn test_alias_fixed_array() {
 	a := [8]byte{init: 22}
 	ret := get(Block(a))
 	println(ret)

--- a/vlib/v/tests/alias_fixed_array_test.v
+++ b/vlib/v/tests/alias_fixed_array_test.v
@@ -1,0 +1,12 @@
+type Block = [8]byte
+
+fn test_alias_fixed_array(){
+	a := [8]byte{init: 22}
+	ret := get(Block(a))
+	println(ret)
+	assert ret == 'Block([22, 22, 22, 22, 22, 22, 22, 22])'
+}
+
+fn get(b Block) string {
+	return '$b'
+}

